File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    count_p = {}
    for player in participant:
        if player in count_p:
            count_p[player] += 1
        else:
            count_p[player] = 1
    
    for player in completion:
        if player in count_p:
            count_p[player] -= 1
            if count_p[player] == 0:
                del count_p[player]
    
    answer = list(count_p.keys())[0]
    
    return answer

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;


class Solution {
    public String solution(String[] participant, String[] completion) {
        Map<String, Integer> comp = new HashMap<>();
        for(String c : participant) comp.put(c, comp.getOrDefault(c, 0) + 1);
        
        for(String p : completion) {
            int r = comp.get(p);
            if(r - 1 == 0) {
                comp.remove(p);
                continue;
            }
            comp.put(p, r - 1);
        }
        String answer = "";
        for(String a : comp.keySet()) {
            return answer = a;
        }
        return answer;
    }
}